Crash with latest Conan 2.0 plugin

As soon as I install the v2.0.0 plugin I got the following error in CLion (2023.2.1):
```
com.intellij.diagnostic.PluginException: Cannot init toolwindow com.jfrog.conan.clion.toolWindow.ConanWindowFactory@23f0e43f [Plugin: com.jfrog.conan.clion]
	at com.intellij.toolWindow.ToolWindowSetInitializerKt.registerToolWindows(ToolWindowSetInitializer.kt:191)
	at com.intellij.toolWindow.ToolWindowSetInitializerKt.access$registerToolWindows(ToolWindowSetInitializer.kt:1)
	at com.intellij.toolWindow.ToolWindowSetInitializer$createAndLayoutToolWindows$entries$1$1.invoke(ToolWindowSetInitializer.kt:129)
	at com.intellij.toolWindow.ToolWindowSetInitializer$createAndLayoutToolWindows$entries$1$1.invoke(ToolWindowSetInitializer.kt:121)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:248)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:199)
	at com.intellij.toolWindow.ToolWindowSetInitializer$createAndLayoutToolWindows$entries$1.invokeSuspend(ToolWindowSetInitializer.kt:121)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:193)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:478)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:79)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:121)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:41)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:685)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$10(IdeEventQueue.kt:589)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithoutImplicitRead(ApplicationImpl.java:1485)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:67)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:369)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:368)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:368)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:363)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:992)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:992)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:363)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:405)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: kotlinx.serialization.MissingFieldException: Field 'libraries' is required for type with serial name 'com.jfrog.conan.clion.models.LibraryData', but it was missing at path: $
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:90)
	at kotlinx.serialization.json.Json.decodeFromString(Json.kt:97)
	at com.jfrog.conan.clion.services.ConanService.getRemoteData(ConanService.kt:233)
	at com.jfrog.conan.clion.toolWindow.ReadmePanel.<init>(ReadmePanel.kt:27)
	at com.jfrog.conan.clion.toolWindow.PackageInformationPanel.<init>(PackageInformationPanel.kt:28)
	at com.jfrog.conan.clion.toolWindow.ConanWindowFactory$ConanWindow.<init>(ConanWindowFactory.kt:43)
	at com.jfrog.conan.clion.toolWindow.ConanWindowFactory.createToolWindowContent(ConanWindowFactory.kt:34)
	at com.intellij.openapi.wm.impl.ToolWindowImpl.createContentIfNeeded(ToolWindowImpl.kt:554)
	at com.intellij.openapi.wm.impl.ToolWindowImpl.scheduleContentInitializationIfNeeded$intellij_platform_ide_impl(ToolWindowImpl.kt:533)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.doShowWindow(ToolWindowManagerImpl.kt:1014)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.showToolWindowImpl(ToolWindowManagerImpl.kt:951)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.showToolWindowImpl$default(ToolWindowManagerImpl.kt:937)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.registerToolWindow$intellij_platform_ide_impl(ToolWindowManagerImpl.kt:1121)
	at com.intellij.toolWindow.ToolWindowSetInitializerKt.registerToolWindows(ToolWindowSetInitializer.kt:181)
	... 43 more
Caused by: kotlinx.serialization.MissingFieldException: Field 'libraries' is required for type with serial name 'com.jfrog.conan.clion.models.LibraryData', but it was missing
	at kotlinx.serialization.internal.PluginExceptionsKt.throwMissingFieldException(PluginExceptions.kt:20)
	at com.jfrog.conan.clion.models.LibraryData.<init>(LibraryData.kt:5)
	at com.jfrog.conan.clion.models.LibraryData$$serializer.deserialize(LibraryData.kt:5)
	at com.jfrog.conan.clion.models.LibraryData$$serializer.deserialize(LibraryData.kt:5)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
	... 56 more

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,9 @@
 
 ## [Unreleased]
 
+### Bugfix
+- Fixed crash due to library data race condition when first using the plugin
+
 ## [2.0.0] - 2023-08-30
 
 ### Added

--- a/gradle.properties
+++ b/gradle.properties
@@ -4,7 +4,7 @@ pluginGroup = com.jfrog.conan.clion
 pluginName = Conan
 pluginRepositoryUrl = https://github.com/conan-io/conan-clion-plugin/
 # SemVer format -> https://semver.org
-pluginVersion = 2.0.0
+pluginVersion = 2.0.1
 
 # Supported build number ranges and IntelliJ Platform versions -> https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
 pluginSinceBuild = 223

--- a/src/main/kotlin/com/jfrog/conan/clion/conan/extensions/file.kt
+++ b/src/main/kotlin/com/jfrog/conan/clion/conan/extensions/file.kt
@@ -7,7 +7,7 @@ import kotlinx.coroutines.launch
 import java.io.File
 import java.net.URL
 
-suspend fun File.downloadFromUrl(url: String) {
+fun File.downloadFromUrl(url: String) {
     val targetFile = this
 
     try {

--- a/src/main/kotlin/com/jfrog/conan/clion/services/ConanService.kt
+++ b/src/main/kotlin/com/jfrog/conan/clion/services/ConanService.kt
@@ -12,7 +12,6 @@ import com.jfrog.conan.clion.conan.ConanPluginUtils
 import com.jfrog.conan.clion.conan.extensions.downloadFromUrl
 import com.jfrog.conan.clion.models.LibraryData
 import com.jfrog.conan.clion.models.PersistentStorageKeys
-import kotlinx.coroutines.runBlocking
 import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.json.Json
 import java.io.File
@@ -159,9 +158,7 @@ class ConanService(val project: Project) {
 
         if (!targetFile.exists() || update && ConanPluginUtils.fileHasOverwriteComment(targetFile)) {
             targetFile.parentFile.mkdirs()
-            runBlocking {
-                targetFile.downloadFromUrl(cmakeProviderURL)
-            }
+            targetFile.downloadFromUrl(cmakeProviderURL)
             // Re-write it, but adding the overwrite header
             ConanPluginUtils.writeToFileWithOverwriteComment(targetFile, targetFile.readText())
             LocalFileSystem.getInstance().refreshAndFindFileByIoFile(targetFile)
@@ -182,20 +179,29 @@ class ConanService(val project: Project) {
 
         if (!targetFile.exists() || update) {
             targetFile.parentFile.mkdirs()
-            runBlocking {
-                targetFile.downloadFromUrl(remoteDataURL)
-            }
+            targetFile.downloadFromUrl(remoteDataURL)
         }
         if (targetFile.exists()) {
             val libraryData = targetFile.readText()
 
-            fireOnLibraryDataChanged(Json{ignoreUnknownKeys=true}.decodeFromString<LibraryData>(libraryData))
+            try {
+                val parsedJson = Json{ignoreUnknownKeys=true}.decodeFromString<LibraryData>(libraryData)
+                fireOnLibraryDataChanged(parsedJson)
+            } catch (e: SerializationException) {
+                thisLogger().error(e)
+                fireOnLibraryDataChanged(LibraryData(hashMapOf()))
+            }
         }
     }
 
     fun getRemoteData(): LibraryData {
-        val targetData = getRemoteDataText()
-        return Json{ignoreUnknownKeys=true}.decodeFromString<LibraryData>(targetData)
+        return try {
+            val targetData = getRemoteDataText()
+            Json { ignoreUnknownKeys = true }.decodeFromString<LibraryData>(targetData)
+        } catch (e: SerializationException) {
+            thisLogger().error(e)
+            LibraryData(hashMapOf())
+        }
     }
 
     fun getRemoteDataText(): String {

--- a/src/main/kotlin/com/jfrog/conan/clion/services/ConanService.kt
+++ b/src/main/kotlin/com/jfrog/conan/clion/services/ConanService.kt
@@ -3,6 +3,7 @@ package com.jfrog.conan.clion.services
 import com.intellij.ide.util.PropertiesComponent
 import com.intellij.openapi.components.Service
 import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.thisLogger
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.ui.Messages
 import com.intellij.openapi.vfs.LocalFileSystem
@@ -12,6 +13,7 @@ import com.jfrog.conan.clion.conan.ConanPluginUtils
 import com.jfrog.conan.clion.conan.extensions.downloadFromUrl
 import com.jfrog.conan.clion.models.LibraryData
 import com.jfrog.conan.clion.models.PersistentStorageKeys
+import kotlinx.serialization.SerializationException
 import kotlinx.serialization.decodeFromString
 import kotlinx.serialization.json.Json
 import java.io.File

